src/lib/ReactDataTable/ReactDataTable.tsx

```diff
@@ -40,167 +46,223 @@ const ReactDataTable = <TData,>(props: ReactDataTableProps<TData>) => {
     }
   }`;
 
+  const DraggableRow = ({ row, table }: { row: Row<TData>; table: Table<TData> }) => {
+    if (!table.options.getRowId) {
+      throw new Error("You must provide 'getRowId()' to data-table options in order to use the drag-and-drop feature.");
+    }
+
+    const { transform, transition, setNodeRef, isDragging } = useSortable({
+      id: row.id,
+    });
+
+    const draggableStyle: CSSProperties = {
+      transform: CSS.Transform.toString(transform),
+      transition: transition,
+      opacity: isDragging ? 0.8 : 1,
+      zIndex: isDragging ? 1 : 0,
+      position: "relative",
+    };
+
+    return (
+      <tr key={row.id} ref={setNodeRef} style={rowStyle ? { ...rowStyle(row.original), ...draggableStyle } : draggableStyle}>
+        {row.getVisibleCells().map((cell) => (
+          <td key={cell.id} style={cell.column.columnDef.meta?.cellStyle}>
+            {flexRender(cell.column.columnDef.cell, cell.getContext())}
+          </td>
+        ))}
+      </tr>
+    );
+  };
+
+  const TableRows = ({ table, dragAndDropOptions }: { table: Table<TData>; dragAndDropOptions?: DragAndDropOptions }) =>
+    dragAndDropOptions?.enableDragAndDrop ? (
+      <SortableContext items={table.getRowModel().rows.map((row) => row.id)} strategy={verticalListSortingStrategy}>
+        {table.getRowModel().rows.map((row, index) => (
+          <DraggableRow key={index} row={row} table={table} />
+        ))}
+      </SortableContext>
+    ) : (
+      <>
+        {table.getRowModel().rows.map((row, index) => (
+          <tr key={index} style={rowStyle && rowStyle(row.original)}>
+            {row.getVisibleCells().map((cell) => (
+              <td key={cell.id} style={cell.column.columnDef.meta?.cellStyle}>
+                {flexRender(cell.column.columnDef.cell, cell.getContext())}
+              </td>
+            ))}
+          </tr>
+        ))}
+      </>
+    );
+
+  const sensors = useSensors(useSensor(MouseSensor, {}), useSensor(TouchSensor, {}), useSensor(KeyboardSensor, {}));
+
   return (
     <>
-      <style>{loadingCss}</style>
-      <ReactStrapTable
-        striped
-        hover
-        size="sm"
-        className={tableClassName}
-        style={
-          !isLoading && isFetching
-            ? {
-                ...tableStyle,
-                background: "linear-gradient(90deg, #E8E8E8, #ffffff, #E8E8E8)",
-                backgroundSize: "200% 200%",
-                animation: "reloadingAnimation 3s linear infinite",
-              }
-            : tableStyle
-        }
+      <DndContext
```

Comment:
if i see it correct, then the context is now always added (even around not draggable tables), do you know if this could have an impact? like performance?

Reply:
As discussed, since we have the disable/enable option runtime, i think we can leave as it is so far. We could discuss afterwards whether the performances look downgraded. Thanks for the useful input ✓